### PR TITLE
ENH: add `styler` option context for sparsification of columns and index separately

### LINKED ISSUE #41142: API: refactor Styler to allow sparse cols or index separately.
The api is inconsistent in some parts. Currently Styler only allows either non-sparse index and cols or sparse index and cols, but not separately. This should be a relatively easy change to implement into the `_translation` and/or `rendering` code.

### PR DESCRIPTION
**Pull request by attack68**
This PR closes #41142.

It adds a new `pandas.options` context for Styler (see #41395), and create the `sparse.index` and `sparse.columns` options, defaulting to True.

It also allows a keyword application for sparsification via the `.render` method, (this might not really be necessary: opinion welcome)

- [x] tests added for kwarg application and option context

### CODE REVIEW

File: doc/source/user_guide/options.rst [hunk 1]
```diff
@@ -482,6 +482,11 @@ plotting.backend                        matplotlib   Change the plotting backend
                                                      like Bokeh, Altair, etc.
 plotting.matplotlib.register_converters True         Register custom converters with
                                                      matplotlib. Set to False to de-register.
+styler.sparsify_index                   True         "Sparsify" MultiIndex display for rows
```
Comment (jreback) [inline doc/source/user_guide/options.rst:485]: we had a discussion a while back about nesting display.styler..... (e.g. as we do display.html), but maybe this was to *move* .html to the top level, this is prob better

consider styler.sparse.index

**Review by jreback — CHANGES_REQUESTED** [review]
didn't look except for first comment

File: doc/source/user_guide/options.rst (hunk 1 above)
Comment (attack68) [reply doc/source/user_guide/options.rst]: i think having its own context is useful for what follows. I have changed to your consideration.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great thanks!

**Pull request merged**